.github/workflows/npm-push.yml

```diff
@@ -0,0 +1,44 @@
+name: NodeJS CI with NPM (Merge)
+
+on:
+  push:
+    branches: [main]
+
+jobs:
+  check-pr-labels:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: read
+    outputs:
+      labels: ${{ steps.get-pr.outputs.pr_labels }}
+    steps:
+    - name: Get Pull Request information
+      uses: 8BitJonny/gh-get-current-pr@2.2.0
+      id: get-pr
+
+  build:
+    runs-on: ubuntu-latest
+    needs: [check-pr-labels]
+    if: contains(needs.check-pr-labels.outputs.labels, 'release')
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 16
+      - run: npm ci
+      #- run: npm test
+    
+  publish-npm:
+    runs-on: ubuntu-latest
+    needs: [build]
+    if: contains(needs.check-pr-labels.outputs.labels, 'release')
```

Comment:
In order for this to work, `needs` should contain `check-pr-labels` as well

Reply:
Question: if it needs the build action to be run, and the build action needs the check-pr-labels actions to be run, is this needed ? There is an implicit dependency.

Reply:
What's the difference between `npm-push` and `npm-publish`? Are we running npm publish twice?

Reply:
@jorgecasar `npm-push` will be executed when a PR is merged to `main`, or a commit is pushed to `main`. `npm-publish` will be executed when a release is created manually on GitHub.

Reply:
> Question: if it needs the `build` action to be run, and the `build` action needs the `check-pr-labels` action to be run, is this needed ? There is an implicit dependency.

It's needed in order to be able to reference its results in the if expression `needs.check-pr-labels.outputs.labels`

Reply:
@5uso-sng done :)